File: erp-app/src/app/modules/accounting/models/account-statement-search.dto.ts
import { AccountTypeEnum } from "src/app/shared/enums/account-type.enum";
import { PaymentMethodEnum } from "src/app/shared/enums/payment-method.enum";
import { PagingDTO } from "src/app/shared/models/paging-dto";

export class AccountStatementSearchDTO extends PagingDTO {
    number: string;
    date: string;
    dateFrom: string | undefined;
    dateTo: string | undefined;
    clientVendorId: number | null;
    phone: string| null;
    beneficiaryName: string;
    paymentMethodId: PaymentMethodEnum | null;
    refNo: string;
    bankAccountNo: string;
    checkNo: string;
    representiveId: number | null;
}

File: ERPServer/Shared/Shared/Entities/Accouting/AccountStatementSearchDTO.cs
using Shared.Entities.Shared;
using Shared.Enums;
using System;

namespace Shared.Entities.Accouting
{
    public class AccountStatementSearchDTO : Paging
    {
        public string DateFrom { get; set; }
        public string DateTo { get; set; }
        public string Date { get; set; }
        public long? ClientVendorId { get; set; }
        public TransactionTypeEnum? TransactionTypeId { get; set; }
        public PaymentMethodEnum? PaymentMethodId { get; set; }
        public string RefNo { get; set; }
        public string BankAccountNo { get; set; }
        public string CheckNo { get; set; }
        public string Notes { get; set; }
        public long? RepresentiveId { get; set; }
        public string PhoneNumber { get; set; }
    }
}
